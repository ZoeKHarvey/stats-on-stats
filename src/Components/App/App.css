.App {
  text-align: center;
  height: 100vh;
  font-family: 'Barlow Condensed', sans-serif;
}

