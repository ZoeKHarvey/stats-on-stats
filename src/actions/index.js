export const getTeams = (teams) => ({
  type:'GET_TEAMS',
  teams
})

export const getStandings = (standings) => ({
  type: 'GET_STANDINGS',
  standings
})

export const getRoster = (roster) => ({
  type: 'GET_ROSTER',
  roster
})

export const getPlayer = (player) => ({
  type: 'GET_PLAYER',
  player
})

export const getPlayerStats = (player) => ({
  type: 'GET_PLAYER_STATS',
  player
})

export const getTeamSchedule = (schedule) => ({
  type: 'GET_TEAM_SCHEDULE',
  schedule
})

export const getPlayerProjections = (player) => ({
  type: 'GET_PLAYER_PROJECTIONS',
  player
})

export const favoriteTeams = (newTeam) => ({
  type: 'GET_FAVORITE_TEAMS',
  newTeam
})

export const isLoading = (isLoading) => ({
  type: 'IS_LOADING',
  isLoading
})

export const hasError = (errorMsg) => ({
  type: 'HAS_ERROR',
  errorMsg
})



