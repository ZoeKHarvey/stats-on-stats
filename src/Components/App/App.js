import { getTeams, isLoading, hasError, getRoster } from '../../actions';
import { bindActionCreators } from 'redux';
import { fetchTeams, fetchRoster } from '../../apiCalls';
import { connect } from 'react-redux';
import React, { Component } from 'react';
import './App.css';
import TeamContainer from '../TeamContainer/TeamContainer';
import { Route } from 'react-router-dom';
import WelcomePage from '../WelcomePage/WelcomePage'
import RosterContainer from '../RosterContainer/RosterContainer';
import StandingsContainer from  '../StandingsContainer/StandingsContainer'

export class App extends Component {
  componentDidMount = async () => {
    const { getTeams, hasError, isLoading } = this.props;
    try {
      const teams = await fetchTeams();
      console.log('teams in app--->', teams)
      getTeams(teams);
    } catch (error) {
      console.log('error')
    }
  }

  getSingleRoster = async(e, id) =>{
    e.preventDefault()
    const { getRoster } = this.props;
    try {
      const roster = await fetchRoster(id);
      getRoster(roster)
    } catch(error) {
      console.log('error')
    }
  }

  getSinglePlayer = async(e, id) => {
    e.preventDefault();
    const { getPlayer } = this.props;
    try {
      const player = await fetchSinglePlayer(id);
      getPlayer(player)
    } catch(error) {
      console.log('error')
    }
  }

  render() {
    return(
      <section className="section-app">
        <Route exact path='/' render={() => <WelcomePage /> } />
        <Route exact path='/teams' render={() => <TeamContainer getSingleRoster={this.getSingleRoster}  />} />
        <Route exact path='/roster' render={() => <RosterContainer />} />
        <Route exact path='/standings' render={() => <StandingsContainer /> } />
      </section>
    )
  }

}

export const mapStateToProps = (state) => ({
  teams: state.teams,
  errorMsg: state.errorMsg,
});

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    getTeams,
    hasError,
    isLoading,
    getRoster
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(App)
