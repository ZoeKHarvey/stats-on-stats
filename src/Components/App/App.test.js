import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { getTeams, getRoster, getPlayer, getTeamSchedule } from '../../actions/index'

jest.mock('../../apiCalls');

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return an object with', () => {
    let teams;
    const mockState = {
      teams
    };

    const expected = {
      teams
    };

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected)
  });
})

  describe('mapDispatchToProps', () => {
    let mockDispatch;

    beforeEach(() => {
      mockDispatch = jest.fn();
    });

    it('calls dispatch with a getTeams action', () => {
      const actionToDispatch = getTeams([{}, {}]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getTeams([{}, {}]);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('calls dispatch with a getRoster action', () => {
      const actionToDispatch = getRoster([{}, {}, {}]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getRoster([{}, {}, {}]);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('calls dispatch with a getPlayer action', () => {
      const actionToDispatch = getPlayer({});
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getPlayer({});
      
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });

    it('calls dispatch with a getTeamSchedule action', () => {
      const actionToDispatch = getTeamSchedule({});
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getTeamSchedule({});

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

  })