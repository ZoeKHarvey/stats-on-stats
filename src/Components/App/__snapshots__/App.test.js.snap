// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<section
  className="section-app"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/teams"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/roster"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/player"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/favorites"
    render={[Function]}
  />
  <Footer />
</section>
`;
